src/controllers/user/me.rs

```diff
@@ -36,12 +38,28 @@ pub fn me(req: &mut dyn Request) -> CargoResult<Response> {
         ))
         .first::<(User, Option<bool>, Option<String>, bool)>(&*conn)?;
 
+    let owned_crates = crate_owners::table
+        .inner_join(crates::table)
+        .filter(crate_owners::owner_id.eq(user_id))
+        .filter(crate_owners::owner_kind.eq(OwnerKind::User as i32))
```

Comment:
Writing up a response to your other comment made me realize that in this spot we also need a filter here that `crate_owners::deleted.eq(false)`. And since I missed remembering 2 filters now, I'm thinking we should have a function for this so that the logic is in one spot that we're sure is correct. Looking around, I see a few other places where we're selecting "crates currently owned by a user, not a team":

- https://github.com/rust-lang/crates.io/blob/36cfbe31b3a81d3a80782299673e3925edde4758/src/controllers/krate/search.rs#L121-L125
- Sort of here: https://github.com/rust-lang/crates.io/blob/36cfbe31b3a81d3a80782299673e3925edde4758/src/models/krate.rs#L397-L404
- https://github.com/rust-lang/crates.io/blob/36cfbe31b3a81d3a80782299673e3925edde4758/src/models/user.rs#L132-L139

Are you up for a little refactoring? Just a little bit of scope creep? 😇

Reply:
I can pull that out into another function...no problem!

Reply:
I just pushed a commit that refactors as mentioned. I hope it's what you're looking for!

Reply:
BTW, I also made a little change to the way the "upsert" works. When building the inserts, I filtered out the records that won't change. Perhaps this is unnecessary based on what you've stated above, but it seems nice to only try to update the records that need updating. I'm fine with whatever, though.

Reply:
> Perhaps this is unnecessary based on what you've stated above, but it seems nice to only try to update the records that need updating. I'm fine with whatever, though.

Yeah I think it's probably unnecessary, but it's also fine :)

Reply:
I removed it for sake of simplicity in the latest commit.